space/load.go

```diff
@@ -10,65 +11,76 @@ import (
 func (s *service) startLoad(ctx context.Context, spaceID string) (err error) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	info, loaderCreated, err := s.createLoaderOrReturnInfo(ctx, spaceID)
-	if err != nil {
-		return
-	}
-	if loaderCreated {
-		err = s.techSpace.SetInfo(ctx, info)
-		if err != nil {
-			return
-		}
-	}
-	return
-}
 
-func (s *service) createLoaderOrReturnInfo(ctx context.Context, spaceID string) (info spaceinfo.SpaceInfo, loaderCreated bool, err error) {
-	currentInfo := s.techSpace.GetInfo(spaceID)
+	status := s.getStatus(spaceID)
 
-	if currentInfo.LocalStatus != spaceinfo.LocalStatusUnknown {
-		// loading already started
-		return currentInfo, false, nil
+	if status.LocalStatus != spaceinfo.LocalStatusUnknown {
+		return nil
 	}
 
-	viewID, err := s.techSpace.DeriveSpaceViewID(ctx, spaceID)
+	exists, err := s.techSpace.SpaceViewExists(ctx, spaceID)
 	if err != nil {
 		return
 	}
+	if !exists {
+		return ErrSpaceNotExists
+	}
 
-	info = spaceinfo.SpaceInfo{
+	info := spaceinfo.SpaceInfo{
 		SpaceID:     spaceID,
-		ViewID:      viewID,
 		LocalStatus: spaceinfo.LocalStatusLoading,
 	}
+	if err = s.setStatus(ctx, info); err != nil {
+		return
+	}
 	s.loading[spaceID] = newLoadingSpace(s.ctx, s.open, spaceID, s.onLoad)
-	loaderCreated = true
 	return
 }
 
 func (s *service) onLoad(spaceID string, sp Space, loadErr error) (err error) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
+
```

Comment:
maybe use `errors.Is` here? It will be less fragile if someone accidentally wrap `loadErr`

Reply:
Yep, also replaced in other places